Move ARM cmdlets from Resources to Resources.Rest proj

vivsriaus opened the pull request:
This is a massive change list that impacts the core ARM cmdlets like ResourceGroups, deployments, features and providers. We're moving these cmdlets which are currently using the old HYAK client into the Commands.Rest project, using the new autorest SDK client. More updates will follow shortly

vivsriaus commented:
@azuresdkci test this please

vivsriaus commented:
@azuresdkci retest this please

markcowl commented:
@vivsriaus Where do we sit with this PR?

vivsriaus commented:
@markcowl I need help in debugging the latest test failures. I'm on-call for ARM this week, so I'm not able to look at this currently, but if you can help me why I keep getting [this ](http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/769/artifact/src/Publish/TestResults/AzurePowershellAllTestResults.html#failures) error, that'd be helpful.

markcowl commented:
@azuresdkci retest this please

vivsriaus commented:
Here is the on-demand job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/785/

Thanks @TianoMS

vivsriaus commented:
@azuresdkci retest this please

gandhiniraj commented on src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1:
```diff
@@ -0,0 +1,158 @@
+  param(
+    [string] [Parameter(Position=0, ValueFromPipelineByPropertyName=$true)] [alias("ResourceGroupName")] $Name,
+    [string] [Parameter(Position=1, ValueFromPipelineByPropertyName=$true)] $Location,
+	[string] [Parameter(ValueFromPipelineByPropertyName=$true)] $Id,
```
nit: aligning

gandhiniraj replied:
nit: e.g. aligning here

gandhiniraj replied:
I don't see $Id is being used below, do we still need get it?

gandhiniraj commented on src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1:
```diff
@@ -0,0 +1,158 @@
+﻿function Get-AzureRmResourceGroup
```
nit: I assume VS doesn't do a good job on aligning PS, but we should maintain that.

gandhiniraj commented on src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1:
```diff
@@ -0,0 +1,158 @@
+	} else {
+	  $getTask = $client.ResourceGroups.GetAsync($Name, [System.Threading.CancellationToken]::None)
+	  $rg = $getTask.Result
+	  $resourceGroup = Get-ResourceGroup $Name $Location $rg.ResourceGroup.Id
```
Do we need to also check if $Location is null?

gandhiniraj commented on src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1:
```diff
@@ -0,0 +1,158 @@
+﻿function Get-AzureRmResourceGroup
+{
+  [CmdletBinding()]
+  param(
```
nit: PARAM to be in sync with BEGIN and PROCESS

gandhiniraj commented on src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1:
```diff
@@ -0,0 +1,158 @@
+  PROCESS {
+    $createParms = New-Object -Type Microsoft.Azure.Management.Resources.Models.ResourceGroup
+	$createParms.Location = $Location
+	#$createParms.Tags = $Tags
```
nit: remove commented code

gandhiniraj commented on src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1:
```diff
@@ -0,0 +1,158 @@
+	$client = Get-ResourcesClient $context
+  }
+  PROCESS {
+    $createParms = New-Object -Type Microsoft.Azure.Management.Resources.Models.ResourceGroup
```
createParams?

vivsriaus commented:
@gandhiniraj Addressed your feedback. Thanks

vivsriaus commented:
Latest on demand job here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/787/

hovsepm commented:
@vivsriaus please resolve merge conflicts

vivsriaus commented:
@hovsepm done

markcowl commented:
on demand job after all changes: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/799/

markcowl commented:
signing job here:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/432/

hovsepm commented:
@vivsriaus the failures in on-demand job is fixed by Batch team and are merged. Please pull from upstream and restart the on-demand after that.

markcowl commented:
@vivciraus - I think the issues are related to merge problems- can you pull the latest from the dev branch?

vivsriaus commented:
@hovsepm @markcowl Here is the new on-demand job. Hopefully, this should do it: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/815/

vivsriaus commented:
@markcowl @hovsepm More changes were merged from upstream dev, so here is the new on-demand job (the previous one succeeded, FWIW): http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/816/

vivsriaus commented:
New on demand job here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/821/

hovsepm commented:
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/436/

vivsriaus commented:
Latest on demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/827/

hovsepm commented:
Failing.playlist  and NewFailing.playlist should not be checked in

hovsepm commented:
@ src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Components\ResourceIdentifier.cs, Line 146: private void AppendIfNotNull(ref StringBuilder resourceId, string format, string value) -> no need to use ref for StringBuilder, you are not changing reference of resourceId in the body anyhow.

hovsepm commented:
@vivsriaus Since this is a huge change in the existing code please include ChangeLog.md documentation with details for end user.

hovsepm commented:
@src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\Features\GetAzureProviderFeatureCmdlet.cs, line 62 and 76 -> please add the string to project resources and use it from there.

hovsepm commented:
in general: WriteWarning("The output object type of this cmdlet will be modified in a future release."); the text string should go to resources. please fix it everywhere.

hovsepm commented:
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\Resource\InvokeAzureResourceActionCmdlet.cs, line: 56 & 57, strings should go to resources as well

hovsepm commented:
files in src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroupDeployments have very long lines which makes it hard to view on github, please limit them to 100~120 symbols per line.

hovsepm commented:
files in src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups have very long lines which makes it hard to view on github, please limit them to 100~120 symbols per line.

hovsepm commented:
in src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups\GetAzureResourceGroupCmdlet.cs, line 79. Instead of catching all the exceptions and silently continuing please write logs at debug stream - aka WriteDebug

hovsepm commented:
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\SdkClient\ResourceManagerSdkClient.cs -> add license header.

hovsepm commented:
the naming convention for PS wrappers of model class is PS[nameOfTheModelToWrap]. you have several classes that violates that. Please fix:
* CreatePSResourceGroupDeploymentParameters.cs
* ValidatePSResourceGroupDeploymentParameters.cs
* CreatePSResourceGroupParameters.cs
* UpdatePSResourceGroupParameters.cs

hovsepm commented:
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Utilities\TemplateUtility.cs - missing license header.

hovsepm commented:
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Utilities\TemplateUtility.cs, line 108. Please add exception details through WriteDebug.

hovsepm commented:
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Utilities\TemplateUtility.cs, line 172. Change Debug.Assert to argument exception

hovsepm commented:
due to huge changes please bump up major version in AzureRM.Resources.psd1

hovsepm commented:
move strings to resources: 
* src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroupDeployments\NewAzureResourceGroupDeploymentCmdlet.cs line 69, 76
* src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups\FindAzureResourceGroupCmdlet.cs line 78
* src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups\NewAzureResourceGroupCmdlet.cs line 46
* \src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups\SetAzureResourceGroupCmdlet.cs line 54
* src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\SdkClient\ResourceManagerSdkClient.cs line 212, 213, 301, 595, 624, 784, 861, 865, 874,

gandhiniraj commented on src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Properties/Resources.resx:
```diff
@@ -117,10 +117,238 @@
+    <value>Are you sure you want to cancel the active deployment of resource group '{0}'?</value>
+  </data>
+  <data name="CancelResourceGroupDeploymentMessage" xml:space="preserve">
+    <value>Cancelling active resource group deployment ...</value>
```
nit: no space after "deployment" and "..." or get line 121 in sync with this line

hovsepm commented:
src\ResourceManager\Resources\Commands.Resources.Test\ScenarioTests\ResourcesController.cs, you should introduce an overload for RunPsTest which takes XunitTracingInterceptor as first parameter and then assigns it to helper.TracingInterceptor (e.g. https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/Profile/Commands.Profile.Test/ProfileController.cs#L57). This will let you have output logs from PS code execution.

vivsriaus commented:
@hovsepm I've addressed all the feedback, except for the last one: Adding the XunitTracingInterceptor to our test controller. I'm not comfortable making this change yet, as I don't know what the impact of this change will be on other RP projects that depend on this. If the above changes are merged, I will then be comfortable working on this tomorrow. Let me know if that sounds good.

vivsriaus commented:
@hovsepm latest on demand job here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/847/

vivsriaus commented:
@hovsepm latest sign job here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/439/

vivsriaus commented:
Latest on demand here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/850/

vivsriaus commented:
Updated on demand here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/853/

hovsepm commented:
@vivsriaus "This branch has conflicts that must be resolved" please pull from upstream and resolve conflicts

vivsriaus commented:
@hovsepm done. Latest on demand here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/854/

vivsriaus commented:
@hovsepm things looking good on this pr?

hovsepm commented:
@vivsriaus reviewing.

hovsepm commented:
sign: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/857/
